src/main.c

```diff
@@ -517,19 +517,37 @@ int main(int argc, char **argv)
 			}
 			if (strcmp(long_options[option_index].name,
 			           "cookie") == 0) {
-				cli_cfg.cookie = strdup(optarg);
+				char *cookie_with_prefix;
+
+				cookie_with_prefix = strdup_with_prefix(optarg, "SVPNCOOKIE=");
+				if (cookie_with_prefix == NULL) {
+					log_error("Could not prepend \"SVPNCOOKIE=\" to the cookie.\n");
+					goto exit;
+				}
+				free(cli_cfg.cookie);
+				cli_cfg.cookie = cookie_with_prefix;
 				break;
 			}
 			if (strcmp(long_options[option_index].name,
 			           "cookie-on-stdin") == 0) {
-				char *cookie = read_from_stdin(COOKIE_SIZE);
+				char *cookie;
+				char *cookie_with_prefix;
 
+				cookie = read_from_stdin(COOKIE_SIZE);
 				if (cookie == NULL) {
-					log_warn("Could not read the cookie from stdin");
-					break;
+					log_error("Could not read the cookie from stdin\n");
+					goto exit;
 				}
+
+				cookie_with_prefix = strdup_with_prefix(cookie, "SVPNCOOKIE=");
+				free(cookie);
+				if (cookie_with_prefix == NULL) {
+					log_error("Could not prepend \"SVPNCOOKIE=\" to the cookie.\n");
+					goto exit;
+				}
+
 				free(cli_cfg.cookie);
-				cli_cfg.cookie = cookie;
+				cli_cfg.cookie = cookie_with_prefix;
```

Comment:
I would keep it simple as well, without checks, focusing on adding support for `SVPNCOOKIE=`:
```c
				char *cookie = read_from_stdin(COOKIE_SIZE);

				if (cookie == NULL) {
					log_warn("Could not read the cookie from stdin");
					break;
				}

				free(cli_cfg.cookie);
				cli_cfg.cookie = strdup_with_prefix(cookie, "SVPNCOOKIE=");
				free(cookie);
```